### PR TITLE
feat: Add support for Terraform 0.13

### PR DESCRIPTION
**Pull request by nweddle**
I propose keeping support for Terraform 11 and for the most recent stable version (13), potentially removing support for 11 in a later PR.

My reasoning has to do with the underlying version of HCL (HashiCorp Configuration Language) used by these versions. Terraform 11 is the last version to support HCL 1, with 13 (and 12) supporting HCL 2. See [1] for more.

This affects Terraform authors in two ways. Not only is there a breaking change in the syntax of Terraform code written for 11 versus that written for 13, there remains a significant difference in the underlying version of the Terraform Plugin SDK used to build Terraform providers targeting those different versions.

Legacy providers that have traditionally worked with versions of Terraform 11 and below will not function with Terraform 13 (or 12) unless the maintainers refactor their plug-in to support the newer version. Not all maintainers have done this yet and we may have need of these older providers, potentially.

There are indeed differences between Terraform 12 and 13[2], however these differences are not as great as those between Terraform 11 and 12[3]. If a Terraform author wishes to develop a new module that supports modern versions, they should skip ahead to 13, thus removing any need of us to generate version 12 modules.

[1] https://github.com/hashicorp/hcl/wiki/HCL-2.0-Transition
[2] https://www.terraform.io/upgrade-guides/0-13.html
[3] https://www.terraform.io/upgrade-guides/0-12.html

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -33,8 +32,8 @@ module.exports = class extends Generator {
         store: true,
         choices: [
           {
-            name: '0.12',
-            value: '12',
+            name: '0.13',
+            value: '13',
           },
           {
             name: '0.11',
```
Comment (tsalright) [inline generators/app/index.js:39]: Why are we leaving version 11 in there? If we removed 12, can we remove 11 too and just force everything to be 13?

File: generators/app/templates/.terraform-version [hunk 2]
```diff
@@ -1,2 +1,2 @@
-<% if (tfVersion == '11') { %>0.11.14<% } -%>
-<% if (tfVersion == '12') { %>0.12.24<% } -%>
+<% if (tfVersion == '11') { %>latest:^0.11<% } -%>
```
Comment (tsalright) [inline generators/app/templates/.terraform-version:1]: Same as previous question. Why remove 12 if we offer 11 and 13. Shouldn't we offer 11, 12, and 13? Or should we just offer 12 and 13? Or just 13?

File: generators/app/templates/examples/main.tf [hunk 3]
```diff
@@ -0,0 +1,8 @@
+provider "null" {}
+
+module "example_module" {
+  source = "../."
+
+  mandatory = "mandatory_string"
+<% if (tfVersion == '11') { %>  input     = "${var.input}"<% } %><% if (tfVersion == '13') { %>  input     = var.input<% } %>
```
Comment (tsalright) [inline generators/app/templates/examples/main.tf:7]: Same about the whole 11, 12, 13 as other comments.

File: generators/app/templates/examples/versions.tf [hunk 4]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_providers {
+    null = {
+      source  = "hashicorp/null"
+      version = "~> 2.1"
+    }
+  }
+
+<% if (tfVersion == '11') { %>  required_version = "< 0.12"<% } %><% if (tfVersion == '13') { %>  required_version = ">= 0.13"<% } %>
```
Comment (tsalright) [inline generators/app/templates/examples/versions.tf:9]: Same as the 11, 12, 13 comments.

File: generators/app/templates/variables.tf [hunk 5]
```diff
@@ -1,10 +1,16 @@
 variable "mandatory" {
-  type        = string
+<% if (tfVersion == '11') { %>  type        = "string"<% } %><% if (tfVersion == '13') { %>  type        = string<% } %>
```
Comment (tsalright) [inline generators/app/templates/variables.tf:2]: 11, 12, 13 terraform support?

**Review by tsalright — CHANGES_REQUESTED** [review]
Everything looks good, but I wonder why we only support 11 or 13 terraform and completely removed 12. Either we support 11, 12, and 13. Or Lastest - 1 (12 and 13). Or just latest (13).

**Comment by nweddle** [comment]
> Everything looks good, but I wonder why we only support 11 or 13 terraform and completely removed 12. Either we support 11, 12, and 13. Or Lastest - 1 (12 and 13). Or just latest (13).

Apologies, my thinking should have been explained in the PR comment, I will add the below to it.

I propose keeping support for Terraform 11 and for the most recent stable version (13), potentially removing support for 11 in a later PR.

My reasoning has to do with the underlying version of HCL (HashiCorp Configuration Language) used by these versions. Terraform 11 is the last version to support HCL 1, with 13 (and 12) supporting HCL 2. See [1] for more.

This affects Terraform authors in two ways. Not only is there a breaking change in the syntax of Terraform code written for 11 versus that written for 13, there remains a significant difference in the underlying version of the Terraform Plugin SDK used to build Terraform providers targeting those different versions.

Legacy providers that have traditionally worked with versions of Terraform 11 and below will not function with Terraform 13 (or 12) unless the maintainers refactor their plug-in to support the newer version. Not all maintainers have done this yet and we may have need of these older providers, potentially.

There are indeed differences between Terraform 12 and 13[2], however these differences are not as great as those between Terraform 11 and 12[3]. If a Terraform author wishes to develop a new module that supports modern versions, they should skip ahead to 13, thus removing any need of us to generate version 12 modules.

[1] https://github.com/hashicorp/hcl/wiki/HCL-2.0-Transition
[2] https://www.terraform.io/upgrade-guides/0-13.html
[3] https://www.terraform.io/upgrade-guides/0-12.html

**Review by tsalright — APPROVED** [review]

**Pull request merged**